x-pack/plugins/lens/public/drag_drop/drag_drop.test.tsx

```diff
@@ -300,219 +306,671 @@ describe('DragDrop', () => {
+          order: [2, 0, 1, 0],
         },
-      },
+      ];
+      const component = mount(
+        <ChildDragDropProvider
+          {...{
+            ...defaultContext,
+            dragging: { ...items[0].value, ghost: { children: <div>Hello</div>, style: {} } },
+            setActiveDropTarget,
+            setA11yMessage,
+            activeDropTarget: { ...items[1].value, onDrop, dropType: 'move_compatible' },
+            dropTargetsByOrder: {
+              '2,0,1,0': { ...items[1].value, onDrop, dropType: 'move_compatible' },
+            },
+            keyboardMode: true,
+          }}
+        >
+          {items.map((props) => (
+            <DragDrop {...props} key={props.value.id}>
+              <div />
+            </DragDrop>
+          ))}
+        </ChildDragDropProvider>
+      );
+
+      expect(component.find(DragDrop).at(1).find('.lnsDragDrop_ghost').text()).toEqual('Hello');
     });
-    expect(setA11yMessage).toBeCalledWith('Lifted hello');
   });
 
-  test('Keyboard navigation: ActiveDropTarget gets ghost image', () => {
+  describe('multiple drop targets', () => {
+    let activeDropTarget: DragContextState['activeDropTarget'];
     const onDrop = jest.fn();
-    const setActiveDropTarget = jest.fn();
+    let setActiveDropTarget = jest.fn();
     const setA11yMessage = jest.fn();
-    const items = [
-      {
-        draggable: true,
-        value: {
-          id: '1',
-          humanData: { label: 'Label1', position: 1 },
+    let component: ReactWrapper;
+    beforeEach(() => {
+      activeDropTarget = undefined;
+      setActiveDropTarget = jest.fn((val) => {
+        activeDropTarget = value as DragContextState['activeDropTarget'];
+      });
+      component = mount(
+        <ChildDragDropProvider
+          setA11yMessage={jest.fn()}
+          dragging={{ id: '1', humanData: { label: 'Label1' } }}
+          setDragging={jest.fn()}
+          setActiveDropTarget={setActiveDropTarget}
+          activeDropTarget={activeDropTarget}
+          keyboardMode={false}
+          setKeyboardMode={(keyboardMode) => true}
+          dropTargetsByOrder={undefined}
+          registerDropTarget={jest.fn()}
+        >
+          <DragDrop
+            value={{ id: '3', humanData: { label: 'ignored' } }}
+            draggable={true}
+            order={[2, 0, 1, 0]}
+          >
+            <button>Drag this</button>
+          </DragDrop>
+          <DragDrop
+            order={[2, 0, 1, 0]}
+            value={value}
+            onDrop={onDrop}
+            dropTypes={['move_compatible', 'duplicate_compatible', 'swap_compatible']}
+            getCustomDropTarget={(dropType) => <div className="extraDrop">{dropType}</div>}
+          >
+            <button>Drop here</button>
+          </DragDrop>
+        </ChildDragDropProvider>
+      );
+    });
+    test('extra drop targets render correctly', () => {
+      expect(component.find('.extraDrop').hostNodes()).toHaveLength(2);
```

Comment:
I actually learnt about this one when I was trying to figure out why `component.find('.extraDrop')` shows 4 elements instead of two. I guess you know that already, but for the rest of reviewers: it's here because when you pass a className (or data-test-subj) to the React component and then the React component passes it to the DOM tree, it's counted as two. (React component with its prop `className` + div that takes the `className`.